src/acrcssc/azext_acrcssc/helper/_ociartifactoperations.py

```diff
@@ -0,0 +1,233 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+"""A module to handle ORAS calls to the registry."""
+# pylint: disable=line-too-long
+# pylint: disable=logging-fstring-interpolation
+import os
+import dataclasses
+import tempfile
+import shutil
+import subprocess
+
+from oras.client import OrasClient
+from azure.cli.core.azclierror import AzCLIError
+from azure.cli.command_modules.acr.repository import acr_repository_delete
+from azure.mgmt.core.tools import parse_resource_id
+from jsonschema.exceptions import ValidationError
+from knack.log import get_logger
+from ._constants import (
+    BEARER_TOKEN_USERNAME,
+    CONTINUOUSPATCH_CONFIG_SCHEMA_V1,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_DRYRUN,
+    CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME,
+    CSSC_WORKFLOW_POLICY_REPOSITORY,
+    SUBSCRIPTION
+)
+from ._utility import (
+    convert_cron_to_schedule,
+    get_task
+)
+
+logger = get_logger(__name__)
+
+
+def create_oci_artifact_continuous_patch(registry, cssc_config_file, dryrun):
+    logger.debug(f"Entering create_oci_artifact_continuouspatching with parameters: {registry.name} {cssc_config_file} {dryrun}")
+
+    oras_client = None
+    temp_artifact_name = None
+    try:
+        oras_client = _oras_client(registry)
+        # we might have to handle the tag lock/unlock for the cssc config file,
+        # to make it harder for the user to change it by mistake
+
+        # the ORAS client can only work with files under the current directory
+        temp_artifact = tempfile.NamedTemporaryFile(
+            prefix="cssc_config_tmp_",
+            mode="w+b",
+            dir=os.getcwd(),
+            delete=False)
+
+        temp_artifact_name = temp_artifact.name
+        user_artifact = open(cssc_config_file, "rb")
+        shutil.copyfileobj(user_artifact, temp_artifact)
+        temp_artifact.close()
+
+        if dryrun:
```

Comment:
Question: we push the config file even in dry run?

Reply:
on dryrun we push  the artifact under a different tag (*:dryrun instead of *:v1) so the task can read the config file from it